tests/main/snapd-state/task.yaml

```
@@ -14,9 +14,21 @@ execute: |
     "$TESTSTOOLS"/snapd-state --foo 2>&1 | MATCH "snapd-state: no such command: --foo"
     "$TESTSTOOLS"/snapd-state foo 2>&1 | MATCH "snapd-state: no such command: foo"
 
+    function check_date() {
+        local TEST_DATE="$1"
+        local TOLERANCE_SECONDS=1800 # 30 minutes
```

5 minutes will probably be sufficient
```suggestion
        local TOLERANCE_SECONDS=300 # 5 minutes
```

---

I wish! :-)

If you look at the logs for the failure, the initial timestam was taken at `2021-09-08T23:53:19+00:00` and the failure was detected at `2021-09-09 00:01:57`, that is almost 9 minutes later. Sure, we could reduce the tolerance, but I don't see a benefit in it (and we risk failures on very congested situations).